### ISSUE TITLE
driver-only ECDSA: add ssl-opt.sh testing with testing parity

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#6861]
**Context:** https://github.com/Mbed-TLS/mbedtls/issues/6839.

This task is to progress TL-ecdsa.TLS by running `ssl-opt.sh` in `component_test_psa_crypto_config_accel_ecdsa_use_psa()` (and the companion reference component) and getting testing parity.

This will likely require updating existing dependency declarations in `ssl-opt.sh` and adding missing declarations, using relevant macros as in #6860.

Strongly consider if it's possible to automatically infer dependencies from the command-line arguments, as is already done for some of them in `detect_required_features()`, rather than manually adding a bunch of declarations on a bunch of test cases.

Note that the dependencies may come not just from the ciphersuite (key exchange), but form the certificates/private keys used as well.

Note: since this modifies dependency declarations, we'll want to use `docs/architecture/psa-migration/outcome-analysis.sh` on the PR that implements this.

Note: enabling more key exchanges in `config_psa_crypto_config_ecdsa_use_psa` is out of scope here, it will be a separate task.

**Depends on:** #6860
